src/quickgui/plugin/qgsquickplugin.h

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************
+  qgsquickplugin.h
+  --------------------------------------
+  Date                 : Nov 2017
+  Copyright            : (C) 2017 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSQUICKPLUGIN_H
+#define QGSQUICKPLUGIN_H
+
+#include <QQmlExtensionPlugin>
+
+/**
+ * \ingroup quick
+ *
+ * Qgis Qml Extension Plugin responsible for exposing C++ Qgis classes to QML
+ *
+ * \since QGIS 3.2
+ */
+class QgisQuickPlugin : public QQmlExtensionPlugin
+{
+    Q_OBJECT
+    Q_PLUGIN_METADATA( IID "org.qt-project.Qt.QQmlExtensionInterface" )
```

Comment:
Should this rather refer to QGIS than to the Qt Project?

Reply:
The `Q_PLUGIN_METADATA` references a Qt interface required for Qt Plugin system. This is  [QmlExtension](http://doc.qt.io/qt-5/qqmlextensionplugin.html) plugin type, so we define here that QgisQuickPlugin is a QML Qt plugin.